src/modules/feature-modules/admin/pages/users/user-new.config.ts

```diff
@@ -1,200 +1,212 @@
-            id: 'unitIds',
-            dataType: 'checkbox-array',
-            label: 'Which unit is the user associated to?',
-            validations: { isRequired: [true, 'Unit is required'] },
-            items: unitsList.sort((a, b) => {
-              const x = a.label.toUpperCase();
-              const y = b.label.toUpperCase();
-              return x === y ? 0 : x > y ? 1 : -1;
-              })
-          }]
-        }),
-      );
+    if (data.organisationId) {
+      const organisation = (data.organisations ?? []).find(o => o.id === data.organisationId);
+      const units = (organisation?.units ?? []).map(u => ({ value: u.id, label: u.name }));
+
+      if (units.length === 1) {
+        data.unitIds = units.map((u) => u.value);
+      } else {
+        steps.push(
+          new FormEngineModel({
+            parameters: [
+              {
+                id: 'unitIds',
+                dataType: 'checkbox-array',
+                label: stepsLabels.l5.label,
+                validations: { isRequired: [true, 'Unit is required'] },
+                items: units.sort((a, b) => a.value.localeCompare(b.value)),
+              },
+            ],
+          })
+        );
+      }
     }
-
   }
-
 }
 
 function inboundParsing(data: InboundPayloadType): StepPayloadType {
   return {
-    email: '',
-    name: '',
-    role: null,
-    organisationId: null,
-    unitIds: null,
-    organisationsList: data.organisationsList
+    contextType: data.contextType ?? 'BASE',
+    ...(data.contextType === 'BASE' && { organisations: data.organisations }),
+    ...(data.contextType === 'UNIT' && { organisations: data.organisations, organisationId: data.organisationId, unitIds: data.unitIds }),
+    ...(data.contextType === 'TEAM' && { role: data.role })
   };
 }
 
 function outboundParsing(data: StepPayloadType): OutboundPayloadType {
-  if (data.role === UserRoleEnum.ADMIN || data.role === UserRoleEnum.ASSESSMENT) {
-    return {
-      email: data.email,
-      name: data.name,
-      role: data.role
-    };
-  }
   return {
-    email: data.email,
-    name: data.name,
-    role: data.role,
-    organisationId: data.organisationId,
-    unitIds: data.unitIds
+    name: data.name ?? '',
+    email: data.email ?? '',
+    role: data.role ?? UserRoleEnum.ASSESSMENT, // This can't happen, since role is required.
+    ...((data.role === UserRoleEnum.ACCESSOR || data.role === UserRoleEnum.QUALIFYING_ACCESSOR) && { organisationId: data.organisationId, unitIds: data.unitIds })
   };
 }
 
-function summaryParsing(data: StepPayloadType, steps: FormEngineModel[]): WizardSummaryType[] {
-
+function summaryParsing(data: StepPayloadType): WizardSummaryType[] {
   const toReturn: WizardSummaryType[] = [];
 
+  let editStepNumber = 1;
+
   toReturn.push(
-    { label: 'Name', value: data.name, editStepNumber: 2  },
-    { label: 'Email', value: data.email, editStepNumber: 1  },
+    { label: 'Email', value: data.email, editStepNumber: editStepNumber++ },
```

Comment:
"Name" must be the first information at summary

Reply:
I changed it to maintain the order that the questions appear for the user, talked to Mariana and is now changed on Figma